src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSBasicDemo.java

```diff
@@ -0,0 +1,172 @@
+    CdfGcsActions.selectFormat(CdapUtils.pluginProp(formatType));
+    CdfGcsActions.skipHeader();
+    CdfGcsActions.getSchema();
+    SeleniumHelper.waitElementIsVisible(CdfGCSLocators.getSchemaButton);
+  }
+
+  @Then("Run and Preview")
+  public void runAndPreview() {
+    CdfStudioActions.runAndPreviewData();
+  }
+
+  @Then("Save and Deploy Pipeline")
+  public void saveAndDeployPipeline() {
+    CdfStudioActions.pipelineName();
+    CdfStudioActions.pipelineNameIp("TestPipeline" + UUID.randomUUID());
+    CdfStudioActions.pipelineSave();
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Run the Pipeline in Runtime")
+  public void runThePipelineInRuntime() throws InterruptedException {
+    CdfPipelineRunAction.runClick();
+  }
+
+  @Then("Wait till pipeline is in running state")
+  public void waitTillPipelineIsInRunningState() {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 1000000);
+    wait.until(ExpectedConditions.or(ExpectedConditions.visibilityOfElementLocated(
+      By.xpath("//*[@data-cy='Succeeded']")), ExpectedConditions.visibilityOfElementLocated(
+      By.xpath("//*[@data-cy='Failed']"))));
+  }
+
+  @Then("Verify the pipeline status is {string}")
+  public void verifyThePipelineStatusIs(String status) {
+    boolean webelement = SeleniumHelper.verifyElementPresent("//*[@data-cy='" + status + "']");
+    Assert.assertTrue(webelement);
+  }
+
+  @Then("Open Logs")
+  public void openLogs() throws FileNotFoundException {
+    CdfPipelineRunAction.logsClick();
+    BeforeActions.scenario.write(CdfPipelineRunAction.captureRawLogs());
+    PrintWriter out = null;
+    try {
+      out = new PrintWriter(BeforeActions.myObj);
+      out.println(CdfPipelineRunAction.captureRawLogs());
+    } finally {
+      if (out != null) {
+        out.close();
+      }
+    }
+  }
+
+  @Then("validate successMessage is displayed")
+  public void validateSuccessMessageIsDisplayed() {
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Click on Advance logs and validate the success message")
+  public void clickOnAdvanceLogsAndValidateTheSuccessMessage() {
+    CdfLogActions.goToAdvanceLogs();
+    CdfLogActions.validateSucceeded();
+  }
+
+  @Then("Enter the BigQuery Properties for table {string}")
+  public void enterTheBigQueryPropertiesForTable(String table) throws InterruptedException, IOException {
+    CdfBigQueryPropertiesActions.enterBigQueryProperties(CdapUtils.pluginProp(table));
+  }
+
+  @Then("Close the GCS Properties")
+  public void closeTheGCSProperties() {
+    CdfGcsActions.closeButton();
+  }
+
+  @Then("Close the BigQuery Properties")
+  public void closeTheBigQueryProperties() {
+    CdfGcsActions.closeButton();
+  }
+
+  @Then("Get Count of no of records transferred to BigQuery in {string}")
+  public void getCountOfRecordsTransferredToBigQueryIn(String table) throws IOException, InterruptedException {
+    //TODO verify in and out count from UI , remove this table records count
```

Comment:
create jira for TODO items and link to them in comments

Reply:
will take this in CDAP-18634